### ISSUE TITLE
Underscore in code cell `filename=` causes an error in LaTeX rendering Missing $

### ISSUE THREAD

**Issue by UncleSamulus** [quarto-dev/quarto-cli#6990]
### Bug description

It seems that when there is an underscore `_` in the filename key of a .qmd cell code, LaTeX cannot render the file correctly, as it expect `_` to be in math mode.

 
 ```text
 compilation failed- error
Missing $ inserted.
<inserted text> 
                $
l.197 \caption{\texttt{my_example.r}}
                                      

You may need to $ $ around an expression in this file.
see notebooks/test.log for more information.
```

### Steps to reproduce

For instance, the following .qmd file would produce the error, with `quarto render`
 
````qmd
---
format: pdf
---

```{.r filename="my_example.r"}
replicate(sample(1:10), 10)
```
````

### Expected behavior

When I write filename with underscore in quarto code cell, I want it to be LaTeX compilable.

### Actual behavior

An underscore in the `filename` directive makes the LaTeX compilation fail.

### Your environment

- IDE: VSCode, from CLI.
- Fedora 38

### Quarto check output

I have updated to latest quarto github release, and the problem is still here apparently:

✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.1: OK
      Dart Sass version 1.55.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.3.450
      Path: /opt/quarto/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.11.5
      Path: /usr/bin/python3
      Jupyter: 5.3.0
      Kernels: ir, julia-1.9, python3

(-) Checking Jupyter engine render....0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.3.1
      Path: /usr/lib64/R
      LibPaths:
        - /home/sortion/R/x86_64-redhat-linux-gnu-library/4.3
        - /usr/lib64/R/library
        - /usr/share/R/library
      knitr: 1.42
      rmarkdown: 2.21

[✓] Checking Knitr engine render......OK

**Comment by mcanouil** [comment]
Thanks for the report!

Could you properly format your example?

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Comment by mcanouil** [comment]
I made a PR that fixes the issue, see: #6991
<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/ec922133-7a35-403f-b0c1-322d0c89777d">
